Automattic/wp-calypso
# Post By Email: add front-end settings for Simple sites

Related to https://github.com/Automattic/dotcom-forge/issues/7210

## Proposed Changes

- moves Post by Email module and settings into their own file
- implements a Simple Site version of those settings utilizing the endpoints being introduced for 7211-gh-Automattic/dotcom-forge

## Why are these changes being made?


* Post by email is currently only available in Calypso for Atomic sites, but [it appears this may not have been an intentional choice](6941-gh-Automattic/dotcom-forge). As work is done on Post by Voice settings, it makes sense to update the nearby Post by Email at the same time.

## Notes

- This PR took a lot of inspiration from @nightnei's Post by Voice work in #90489. Before I decided to split these settings off into a separate file I actually branched off of that PR so I could more easily keep any changes in sync, so we actually have at least one commit in common at the moment. I'll rebase off of that PR periodically to keep on the same page.
- Currently in a fairly raw state, it needs a lot of cleanup. I liberally applied the props I needed to silence TypeScript, and will now be going through and trimming them down to what's actually needed
- The settings are (naturally) not functional as of yet, that'll wait until the new endpoints are in place
- For the setting toggle, I'm currently rendering either a `JetpackModuleToggle` as the setting did previously, or just a `ToggleControl` from @wordpress/components. I'd like to look into whether or not we can use the `ToggleControl` in both cases and just modify its handlers for Atomic/Jetpack vs Simple sites.

## Testing Instructions

- On your sandbox, apply the patch from 7211-gh-Automattic/dotcom-forge
- Point public-api.wordpress.com to your sandbox
- Running Calypso locally off of this branch, visit **Settings > Writing** for a simple site, and make sure you have the **Classic View** selected (Calypso, not wp-admin)
- Confirm that the Post by Email settings appear
- In a separate tab, open https://dashboard.wordpress.com/
- Find your simple test site in the dashboard and note the state and/or email address of the current post by email settings
- Back in Calypso, test the Post by Email toggle, refreshing and checking dashboard.wordpress.com to confirm the settings from Calypso are carrying over.
- Back in Calypso, test the Regenerate button. Confirm that a new address is generated, and that the new address if reflected when refreshing dashboard.wordpress.com
- Switch over to **Settings > Writing** for a WoA site
- Repeat the same tests, this time comparing them to `wp-admin/admin.php?page=jetpack#/writing`, making sure the Calypso settings still properly update the Jetpack settings (we didn't make changes here, but the code was moved and refactored so we're making sure nothing got broken in the process)
